packages/stg-evm-v2/ts-src/utils/helpers.ts

```diff
@@ -21,3 +26,136 @@ export const appendDependencies =
     (deploy: DeployFunction): DeployFunction => (
         (deploy.dependencies = [...(deploy.dependencies ?? []), ...dependencies]), deploy
     )
+
+/**
+ * Helper function that replaces any $ placeholders in str with the provided address
+ * @param str The string with placeholders
+ * @param address The address to replace the placeholders with
+ * @returns str with placeholders replaced with address
+ */
+export const fillAddress = (str: string, address: string) => {
+    // Ensure the address is 40 characters long (without the 0x prefix)
+    if (address.slice(0, 2) === '0x') {
+        address = address.slice(2)
+    }
+    if (address.length !== 40) {
+        throw new Error(`Invalid library address length ${address}`)
+    }
+
+    // Find the placeholder in the bytecode
+    const placeholder = `$`
+    while (str.includes(placeholder)) {
+        str = str.replace(placeholder, address)
+    }
+
+    return str
```

Comment:
Or something like this:

```typescript
const normalizedAddress = address.replace(/^0x/, '')
assert(normalizedAddress.length === 40, `Invalid library address length for ${address}`)

return str.replaceAll('$', normalizedAddress)
```

This way you avoid mutating `str` and save a couple of lines as well

Reply:
Nice! I don't think `replaceAll` will work without changing the `tsconfig.json` to have `"target": "es2021"` so I used `replace` with a regular expression to replace all instances!